examples/integrations/gcp/src/jobs/etl/images/bounding_box/bounding_box_etl.py

```diff
@@ -19,47 +25,87 @@
 VERTEX_MIN_TRAINING_EXAMPLES = 1
 
 
-def bounding_box_etl(project: Project) -> str:
+def download_image(image_url: str,
+                   scale_w: float = 1 / 4.,
+                   scale_h: float = 1 / 4.) -> Tuple[Image, Tuple[int, int]]:
+    im = load_image(BytesIO(requests.get(image_url).content))
+    w, h = im.size
+    return im.resize((int(w * scale_w), int(h * scale_h))), (w, h)
+
+
+def image_to_bytes(im: Image) -> BytesIO:
+    im_bytes = BytesIO()
+    im.save(im_bytes, format="jpeg")
+    im_bytes.seek(0)
+    return im_bytes
+
+
+def upload_to_gcs(image_bytes: BytesIO, data_row_uid: str,
+                  bucket: Bucket) -> str:
+    # Vertex will not work unless the input data is a gcs_uri
+    gcs_key = f"training/images/{data_row_uid}.jpg"
+    blob = bucket.blob(gcs_key)
+    blob.upload_from_file(image_bytes, content_type="image/jpg")
+    return f"gs://{bucket.name}/{blob.name}"
+
+
+def process_label(label: Label, bucket: Bucket) -> str:
+    bounding_box_annotations = []
+    # TODO: Only download if the label has annotations
```

Comment:
Also, are you downloading even if the image was in gs:// already? Won't that make training with images that were already in a bucket unnecessarily slow/expensive?
I'm not sure how common is for Labelbox projects to have their image in Google buckets so this might not be super relevant.

Reply:
The data can't be in gs:// format because labelbox will only expose a signed url to customers. I agree that we don't want to keep processing the same data but I was going to follow up and maybe make accommodating backend changes.

This runs in a few seconds on my dataset of 250 images so the run time isn't a problem.

Reply:
Oh I see, that makes sense.